Error While Compiling

Hi, I am trying to setup a ble lamp with your code, but I get an error when trying to compile it.
Here is my .yaml file and the error.

Error:

`src/esphome/components/ble_adv_controller/ble_adv_handler.cpp: In member function 'bool esphome::bleadvcontroller::BleAdvHandler::identify_param(const esphome::bleadvcontroller::BleAdvParam&, bool)':
src/esphome/components/ble_adv_controller/ble_adv_handler.cpp:256:115: error: expected primary-expression before ';' token
       nodiff ? ESP_LOGI(TAG, "Decoded / Re-encoded with NO DIFF") : ESP_LOGE(TAG, "DIFF after Decode / Re-encode");
                                                                                                                   ^
*** [.pioenvs/slzb-06-zigbee_bluetooth-e9c5b0/src/esphome/components/ble_adv_controller/ble_adv_handler.cpp.o] Error 1`

This is my Config:

```
substitutions:
  name: slzb-06-zigbee_bluetooth-e9c5b0

packages:
  SMLIGHT.SLZB-06*(Zigbee+Bluetooth): github://smlight-dev/slzb-06-fw-esphome/configs/zb-bt.yaml

esphome:
  name: ${name}
  name_add_mac_suffix: false

api:
  encryption:
    key: ***

external_components:
  - source:
      type: git
      url: https://github.com/aronsky/esphome-components
      ref: main

#esp32_ble_tracker:

ble_adv_controller:
  - id: my_controller
    encoding: lampsmart_pro

light:
  - platform: ble_adv_controller
    ble_adv_controller_id: my_controller
    name: Kitchen Light

button:
  - platform: ble_adv_controller
    ble_adv_controller_id: my_controller
    name: Pair
    cmd: pair
```

--- a/components/ble_adv_controller/ble_adv_handler.cpp
+++ b/components/ble_adv_controller/ble_adv_handler.cpp
@@ -252,8 +252,11 @@ bool BleAdvHandler::identify_param(const BleAdvParam & param, bool ignore_ble_pa
       encoder->encode(params, cmd, cont);
       BleAdvParam & fparam = params.back();
       ESP_LOGD(TAG, "enc - %s", esphome::format_hex_pretty(fparam.get_full_buf(), fparam.get_full_len()).c_str());
-      bool nodiff = std::equal(param.get_const_data_buf(), param.get_const_data_buf() + param.get_data_len(), fparam.get_data_buf());
-      nodiff ? ESP_LOGI(TAG, "Decoded / Re-encoded with NO DIFF") : ESP_LOGE(TAG, "DIFF after Decode / Re-encode");
+      if (std::equal(param.get_const_data_buf(), param.get_const_data_buf() + param.get_data_len(), fparam.get_data_buf())) {
+        ESP_LOGI(TAG, "Decoded / Re-encoded with NO DIFF");
+      } else {
+        ESP_LOGE(TAG, "DIFF after Decode / Re-encode");
+      }
 
       return true;
     } 

